### PR TITLE
Deprecate storage addon

### PR DESCRIPTION
**Pull request by joedborg**
Here we're starting the process of deprecating the `storage` addon and replace it with the `hostpath-storage` addon.

We're also acknowledging the limitations of hostpath storage.

### CODE REVIEW

**Review by balchua — COMMENTED** [review]
IIRC We need to modify `portainer` addon as it is enabling the `storage` addon.
Adding @balasu in the loop, in case he recalls.

**Comment by joedborg** [comment]
Thanks @balchua, have updated that here https://github.com/ubuntu/microk8s/pull/2833/files#diff-3330b8013f6cdb25fbf80b56147a0f8246f1a993bb27b3d7e38bb9ee4819a67dR13-R14

**Comment by balchua** [comment]
Oh the registry also uses the `storage` addon.

File: microk8s-resources/actions/disable.storage.sh [hunk 1]
```diff
@@ -4,20 +4,7 @@ set -e
 
 source $SNAP/actions/common/utils.sh
 
-echo "Disabling default storage"
-read -ra ARGUMENTS <<< "$1"
```
Comment (ktsakalozos) [inline microk8s-resources/actions/disable.storage.sh:8]: What happened to the argument here? It is passed to the new addon?

**Comment by ktsakalozos** [comment]
Should we also update the addons list https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/wrappers/addon-lists.yaml ?

**Comment by joedborg** [comment]
> Should we also update the addons list https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/wrappers/addon-lists.yaml ?

Yes!

**Review by neoaggelos — APPROVED** [review]
LGTM

**Review by balchua — COMMENTED** [review]
Should we consider this issue? I.e. `RemoveSelfLink=false` being deprecated.

https://github.com/ubuntu/microk8s/issues/2829

**Comment by ktsakalozos** [comment]
> Should we consider this issue? I.e. `RemoveSelfLink=false` being deprecated.

Yes, this will be handled by a separate PR.

**Pull request merged**